runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -240,15 +285,55 @@ void registerTable(@Nonnull String namespace,
         }
     }
 
+    /**
+     * For each protobuf filename -> descriptor map, validate if the previous descriptor matches
+     * If there is a new schema or an update, record the new update in the table.
+     * (Note that oldProto means existing schema previously opened)
+     * WARNING: This method MUST be invoked within a transaction.
+     *
+     * @param protoName - name of the protobuf file
+     * @param newProtoFd - descriptor of the protobuf file that is being inserted.
+     * @param namespace - namespace of the table
+     * @param tableName - tablename without the namespace
+     */
+    private void recordNewSchema(ProtobufFileName protoName,
+                                    CorfuRecord<ProtobufFileDescriptor, TableMetadata> newProtoFd,
+                                    String namespace,
+                                    String tableName) {
+        final CorfuRecord<ProtobufFileDescriptor, TableMetadata> oldProto = this.protobufDescriptorTable.get(protoName);
+        if (oldProto == null) {
+            this.protobufDescriptorTable.put(protoName, newProtoFd);
+            return;
+        }
+        if (!oldProto.getPayload().getFileDescriptor().equals(newProtoFd.getPayload().getFileDescriptor())) {
+            if (!protoName.getFileName().startsWith("google/protobuf")) {
+                log.warn("registerTable: Schema update detected for table {}${} in schema file {}",
+                        namespace, tableName, protoName);
+
+                StringBuilder stackTraceStr = new StringBuilder();
+                for (StackTraceElement st : Thread.currentThread().getStackTrace()) {
+                    stackTraceStr.append(st);
+                    stackTraceStr.append("\n");
+                }
+                log.debug("{}", stackTraceStr);
+                log.debug("registerTable: old schema: {}", oldProto.getPayload().getFileDescriptor());
+                log.debug("registerTable: new schema: {}", newProtoFd.getPayload().getFileDescriptor());
+            }
+            this.protobufDescriptorTable.put(protoName, newProtoFd);
```

Comment:
shouldnt this be treated as an error or unexpected case?

Reply:
No, because in data-migration and build-to-build experiments done by developers this is the normal path and we should allow it to happen.